lib/js_rails_routes/generator.rb

```diff
@@ -57,15 +62,27 @@ def handle_route(route_name, route_path)
       "export function #{route_name}_path(params) { return process('#{route_path}', params, [#{keys.join(',')}]); }"
     end
 
-    def routes
-      @routes ||= Rails.application.routes.routes
-                       .select(&:name)
-                       .map { |r| [r.name, r.path.spec.to_s.split('(')[0]] }
-                       .sort { |a, b| a[0] <=> b[0] }
+    def routes_with_engines
+      @routes_with_engines ||=
+        Rails::Engine.subclasses
+                     .reject { |klass| klass.to_s.downcase =~ exclude_engines }
+                     .map { |engine| [engine.to_s, make_routes(engine.routes.routes)] }
+                     .reject { |_, routes| routes.empty? } +
+        [['Rails', make_routes(Rails.application.routes.routes)]]
+    end
+
+    def make_routes(routes)
+      routes
+        .select(&:name)
+        .map { |r| [r.name, r.path.spec.to_s.split('(')[0]] }
+        .sort { |a, b| a[0] <=> b[0] }
     end
 
-    def write(string)
-      File.open(path, 'w') { |f| f.write(string) }
+    def write(rails_engine_name, string)
+      File.open(
+        File.join(base_path, "#{rails_engine_name.gsub('::Engine', '').downcase}-routes.js"),
+        'w'
+      ) { |f| f.write(string) }
```

Comment:
It's ok, but `IO.write` may improve readability IMO.

```rb
file_name = File.join(base_path, "#{rails_engine_name.gsub('::Engine', '').downcase}-routes.js")
File.write(file_name, string)
```

How do you think?

Reply:
Agreed. I fixed that.